Use task_schedule3.pl in the repo

jeanconn opened the pull request:
Use task_schedule3.pl in the repo

This also removes the Makefile hack.

taldcroft commented on task_schedule3.pl:
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env /proj/sot/ska/bin/perl
+#!/usr/bin/env perlska
```
If the Ska environment (and PATH) are not already defined at this point, then `#!/usr/bin/env perlska` is undefined (i.e. no guarantee it will be found).

jeanconn replied:
OK.  Yeah, that's a good point.  So the best thing is to have a full path to a conda installed perlska when installed.  This is fine from a conda perspective, but what should live as the shebang in the git repo?

jeanconn replied:
Alternatively, I think we could define PATH in the crontab, but that's less self-documenting.

taldcroft replied:
`acis_taco` is the template here.  Cron tab has
```
/proj/sot/ska3/flight/bin/skare task_schedule3.pl ...`
```
So `/proj/sot/ska3/flight/bin/skare` is universal way to get into Ska3 and to set PATH so `$SKA/bin/task_schedule3.pl` is found.  Any perl or Python share task scripts should use `/usr/bin/env perl` or `/usr/bin/env python`.

taldcroft commented on Makefile:
```diff
@@ -66,21 +66,9 @@ test_send_mail_quiet: check_install $(TEST_DEP) install
-	chmod +x task_schedule3.pl
 	cp -p task_schedule3.pl $(SKA_ARCH_OS)/bin/
-	rm task_schedule3.pl
+	chmod +x $(SKA_ARCH_OS)/bin/task_schedule3.pl
```
`SKA_ARCH_OS` is deprecated in favor of `SKA_ROOT`.

taldcroft replied:
(But this will work, it's just not preferred for new-ish stuff).

jeanconn replied:
OK.

jeanconn replied:
I can probably cut the chmod anyway, as it isn't a newly created file anymore and should keep the executable status from the repo.

taldcroft replied:
Agreed on cutting.

jeanconn replied:
Also, if `SKA_ARCH_OS` is deprecated, where is `SKA_ROOT` set?  If that's what you want, we should probably add it to `flt_envs` (I don't see it).  INSTALL_BIN would probably work here too as we're still importing the Makefile.FLIGHT.

jeanconn replied:
I set the Makefile basically back to the original install target, as it makes the most sense to do the conda build with just a cp at this point, now that task_schedule3.pl is already "built".

jeanconn commented on task_schedule3.pl:
```diff
@@ -322,7 +322,8 @@ sub check_outputs {
 ##***************************************************************************
     my $cronjob = shift;
 
-    my $watch_config = io(File::Temp::tmpnam);
```
Looks like the fix I had to use File::Temp::tmpnam instead of POSIX::tmpnam isn't working because File::Temp::tmpnam is context sensitive and I think it is defaulting to list context when called in io().

jeanconn commented:
@taldcroft  I think I addressed your comments.  Do you have other things you'd like done in this PR?  My thought was that it really doesn't matter what the Makefile here is doing anymore as it makes the most sense to just use "cp" in the conda repo to just install the application.
